Define link decoration, navigational tracking, and bounce tracking.

jyasskin opened the pull request:
This also adds a rough introduction.


<!--
    This comment and the below content is programatically generated.
    You may add a comma-separated list of anchors you'd like a
    direct link to below (e.g. #idl-serializers, #idl-sequence):
#intro, #terminology
    Don't remove this comment or modify anything below this line.
    If you don't want a preview generated for this pull request,
    just replace the whole of this comment's content by "no preview"
    and remove what's below.
-->
***
<a href="https://pr-preview.s3.amazonaws.com/jyasskin/nav-tracking-mitigations/pull/2.html" title="Last updated on Sep 21, 2021, 12:31 AM UTC (4d600b4)">Preview</a> (<a href="https://pr-preview.s3.amazonaws.com/jyasskin/nav-tracking-mitigations/pull/2.html#intro" title="#intro">#intro</a>) (<a href="https://pr-preview.s3.amazonaws.com/jyasskin/nav-tracking-mitigations/pull/2.html#terminology" title="#terminology">#terminology</a>) | <a href="https://pr-preview.s3.amazonaws.com/privacycg/nav-tracking-mitigations/2/85894c8...jyasskin:4d600b4.html" title="Last updated on Sep 21, 2021, 12:31 AM UTC (4d600b4)">Diff</a>

jyasskin committed:
Define link decoration, navigational tracking, and bounce tracking.

jyasskin committed:
Add a rough introduction.

pes10k commented on index.bs:
```diff
@@ -15,19 +15,107 @@ Complain About: accidental-2119 yes, missing-example-ids yes
+
+:: **Not** decoration: changing the number changes which bug the user sees.
+
+: <code>https://www.google.com/maps/@<em>37.4220328,-122.0847584,17.12z</em></code>
```
I suggest removing this example, or at least describing it differently, as it conflates two things.   Changing the numbers could change the content of the destination page _and also uniquely describe the user_.

jyasskin replied:
I think it's an important example for demonstrating what's hard about navigational-tracking mitigations, but I'm happy to describe it differently.

In this case, the numbers do not encode user-identifying information, and modifying them to embed a user ID wouldn't successfully communicate a user ID to anyone (since nobody's listening within google.com/maps to decode the user ID). But it's hard for an automated system inside a browser to prove that, and even hard for humans reading the URL to be confident of it.

Is that the better description? That there's enough information here (10+ digits) to identify a user, but it's used to identify a place instead, and removing enough digits to prevent it from identifying a user would prevent it from serving its purpose of identifying the right place?

pes10k replied:
What would you think of calling it an "ambiguous" case, with additional text similar to what you've written above?

johannhof commented on index.bs:
```diff
@@ -15,19 +15,107 @@ Complain About: accidental-2119 yes, missing-example-ids yes
+
+<dfn>Link decoration</dfn> is when the source of a [=hyperlink=] "decorates" its [=URL=]
+with extra information beyond what's necessary to identify the page a user wants
+to navigate to. This information can be placed almost anywhere inside the URL.
```
What about cases like https://example.com/login?returnto=item/12345 or https://example.com/auth/callback?token=1234567? By your definition, would these fall in the "necessary to identify the page" category or be acceptable link decoration? I guess I'm wondering if we should extend this definition to include parameters that "enable user-exposed functionality on the page" or something like that.

johannhof replied:
> I feel like those are both (acceptable) link decoration, but the first (https://example.com/login?returnto=item/12345) seems "necessary to identify the page a user wants to navigate to" and so wouldn't be counted as link decoration under my definition. That could indicate a problem with the definition or a problem with my intuition. 🙃 Do you have a sense for which it is?
> 
> The second (https://example.com/auth/callback?token=1234567) is necessary for the page to work, but that can be true for any decoration with the cooperation of the target page, so I want to ensure that the definition can include such cases. The token doesn't identify which page the user wants to see, so I think it counts as "extra information" in the current definition, and so is link decoration. That seems correct to me.
> 
> I've added these to the examples, but I haven't filed github issues for them yet, pending your thoughts.

I agree with you that both are cases of link decoration by your current definition, and it might make sense to keep that definition broader for now.

I wanted to see if we could narrow down the definition of link decoration to the point where any "acceptable" use of parameters is excluded. Thinking about writing browser policy against navigational tracking, it would be great if we had a way to define it without having to carve out cases like auth callback or unsubscribe where it fits the definition but is still somehow acceptable.

But you're right that theoretically any decoration can be constructed to provide some functionality on the site, so I don't have a good solution for this, either. I'm fine with discussing this problem ("can we define acceptable use of navigational tracking?") separately.

jyasskin replied:
One thing I'm trying to do with these definitions is to make "link decoration" the sometimes-acceptable thing, and "navigational tracking" be the never-acceptable thing. Your login-related examples do explore some edge cases in even making navigational tracking never-acceptable. I've filed #8 to discuss that further, but I hope to merge these rough definitions before completely figuring that out.

johannhof commented on index.bs:
```diff
@@ -15,19 +15,107 @@ Complain About: accidental-2119 yes, missing-example-ids yes
+[[HTML#browsing-the-web|navigation]] to identify that a user on one site is the
+same person as a user on another site. Navigations are usually visible to users,
+but it's possible for a site to navigate away and redirect back fast enough that
+only attentive users would notice.
```
This definition sounds a bit like navigational tracking always employs fast navigations for tracking (like bounce tracking), though IIUC we're looking at more than that. Maybe "general use of a navigation or the information that is transmitted across sites as part of a navigation"?

johannhof replied:
Thanks, that seems good!

johannhof commented on index.bs:
```diff
@@ -15,19 +15,107 @@ Complain About: accidental-2119 yes, missing-example-ids yes
+
+: <code>https://publisher.example/page?userId=<em>5789rhkdsaf8urfnsd</em></code>
+
+:: [=Link decoration=], and also [=navigational tracking=].
```
To throw another edge case in here (doesn't have to be addressed in this PR), what about 

publisher.example/**unsubscribe**?userId=5789rhkdsaf8urfnsd

We've seen two different variations of these, one where the userid is only pre-filled into a free-form text input field, the other where the userId is strictly required to unsubscribe.

johannhof replied:
Thanks! As mentioned I think this is a subset of the "how do we define acceptable use of navigational tracking" problem.

jyasskin committed:
Improve the navigational tracking definition and add some more-contentious examples.

jyasskin committed:
Tighten descriptions of the two examples with unfiled issues.

The login URL becomes clearly decoration if the user winds up seeing a login
page.

jyasskin committed:
Use an HTML header instead of Markdown.

jyasskin committed:
Link navigational tracking to its definition in the 'deployed mitigations' section.

IBRAHIM90990 commented:
1.February. 2025.thailand government lottery first price win number.

IBRAHIM90990 commented:
1.February. 2025.thailand government lottery first prices win number 6.digit.

makadjikaramoko828-arch reviewed (commented):
10000000

makadjikaramoko828-arch commented:
![20250902_232256](https://github.com/user-attachments/assets/81cc6aee-71b9-4983-aa87-b5a748508a4b)

makadjikaramoko828-arch commented:
20000

Dubon1966 reviewed (commented):
help freezes buffering and running slow

hunter-Mike-AI reviewed (changes requested):
Revisión
